core/src/cats/make_catalog_backup.pl.in

```diff
@@ -1,154 +1,42 @@
-    if ($args{db_socket}) {     # unix socket is fastest than net socket
-        $addr = "socket=$args{db_socket}";
-    }
-
-    print MY "[client]
-$addr
-user=$args{db_user}
-password=\"$args{db_password}\"
-";
-    if ($args{db_port}) {
-        print MY "port=$args{db_port}\n";
-    }
-
-    close(MY);
-
-    exec("HOME='$wd' mysqldump -f --opt $args{db_name} > '$wd/$args{db_name}.sql'");
-    print "Error while executing mysql dump $!\n";
-    return 1;
-}
-
-sub dump_catalog
-{
-    my %args = @_;
-    if ($args{db_type} eq 'SQLite3') {
-        $ENV{PATH}="@SQLITE_BINDIR@:$ENV{PATH}";
-        dump_sqlite3(%args);
-    } elsif ($args{db_type} eq 'PostgreSQL') {
-        $ENV{PATH}="@POSTGRESQL_BINDIR@:$ENV{PATH}";
-        dump_pgsql(%args);
-    } elsif ($args{db_type} eq 'MySQL') {
-        $ENV{PATH}="@MYSQL_BINDIR@:$ENV{PATH}";
-        dump_mysql(%args);
-    } else {
-        die "This database type isn't supported";
-    }
-}
-
-open(FP, "$dir_conf -C '$cat'|") or die "Can't get catalog information $@";
-# catalog=MyCatalog
-# db_type=SQLite
-# db_name=regress
-# db_driver=
-# db_user=regress
-# db_password=
-# db_address=
-# db_port=0
-# db_socket=
-my %cfg;
-
-while(my $l = <FP>)
-{
-    if ($l =~ /catalog=(.+)/) {
-        if (exists $cfg{catalog} and $cfg{catalog} eq $cat) {
-            exit dump_catalog(%cfg);
-        }
-        %cfg = ();              # reset
-    }
-
-    if ($l =~ /(\w+)=(.+)/) {
-        $cfg{$1}=$2;
-    }
-}
-
-if (exists $cfg{catalog} and $cfg{catalog} eq $cat) {
-    exit dump_catalog(%cfg);
-}
-
-print "Can't find your catalog ($cat) in director configuration\n";
-exit 1;
+#!/bin/sh
+#
+# BAREOS® - Backup Archiving REcovery Open Sourced
+#
+# Copyright (C) 2000-2011 Free Software Foundation Europe e.V.
+# Copyright (C) 2013-2022 Bareos GmbH & Co. KG
+#
+# This program is Free Software; you can redistribute it and/or
+# modify it under the terms of version three of the GNU Affero General Public
+# License as published by the Free Software Foundation and included
+# in the file LICENSE.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+#
+# This script is a wrapper replacing deprecated original perl script
+#
+# $1 is the Catalog Name resource you want to backup
+set +e
+set +u
+
+#
+# Source the Bareos config functions.
+#
+. @scriptdir@/bareos-config-lib.sh
+
+warn "This script is deprecated and will be remove in future version 23, migrate your configuration to use make_catalog_backup"
+
+if [ $# -ne 1 ]; then
+    warn "Script signature has changed: usage $0 CatalogName"
```

Comment:
No, the Script signature has not changed, so this `if` statement is probably not needed at all. Maybe a simple
`exec @scriptdir@/make_catalog_backup "$1"` is enough?